File: OLSource1/v1/VS_officedev/codesnippet/CSharp/ribbon-object-model-overview_1.cs
        public Ribbon1()
            : base(Globals.Factory.GetRibbonFactory())
        {
            InitializeComponent();
            if (System.DateTime.Now.Hour > 16)
            {
                Microsoft.Office.Tools.Ribbon.RibbonButton button =
                    this.Factory.CreateRibbonButton();
                button.Label = "New Button";
                group1.Items.Add(button);
            }
        }

File: OLSource1/v1/VS_officedev/codesnippet/VisualBasic/ribbon-object-model-overview_1.vb
    <System.Diagnostics.DebuggerNonUserCode()> _
    Public Sub New()
        MyBase.New(Globals.Factory.GetRibbonFactory())

        'This call is required by the Component Designer.
        InitializeComponent()
        Dim MyButton As Microsoft.Office.Tools.Ribbon.RibbonButton = _
            Me.Factory.CreateRibbonButton()
        MyButton.Label = "New Button"
        If System.DateTime.Now.Hour > 16 Then
            Group1.Items.Add(MyButton)
        End If

    End Sub